lib/src/facts/posix/networking_resolver.cc

```diff
@@ -120,4 +120,32 @@ namespace facter { namespace facts { namespace posix {
         return result;
     }
 
+    void networking_resolver::assign_address(ipaddress& ip, sockaddr const* address, sockaddr const* mask) const
+    {
+        if (!address) {
+           return;
+        }
+
+        string* assigned = nullptr;
+        bool (*ignored)(string const&) = nullptr;
```

Comment:
Not really.  I generally just use raw pointers when pointing at one of (potentially) many objects to manipulate based on some conditional logic.  I have no preference either way and can easily change that to a `std::function` for readability.